### ISSUE TITLE
Shapes from library added during session not visible to other participants.

### ISSUE THREAD

**Issue by kassiebowman** [excalidraw/excalidraw#2822]
If a session participant adds a shape library during a session, no shapes from that library will be visible to other participants. It also prevents that participant from viewing any shapes from the same library added by others (which may be a separate bug).

Here are the steps to reproduce, along with a gif of the issue:
1. The first user starts a drawing and adds a shape library (e.g. stick figures).
2. Add a shape from this library to the drawing.
3. Start a collaboration session. The second user is able to see the first shape.
4. The first user adds a second shape from the same library. Both users are able to see both shapes and interact with them.
5. The second user adds the same shape library to their drawing.
6. The first user adds another shape from the same library. The second user cannot see the new shape.
7. The second user adds a shape from the same library. The first user cannot see the new shape.

![shape library bug gif](https://user-images.githubusercontent.com/19256422/105086547-00111300-5a4e-11eb-9829-89a1385b0e27.gif)

**Comment by dwelle** [comment]
Checking your gif, it looks like the case where if doesn't sync with the second client is because you're no longer working in the collaboration room. This happened because by importing the library, it opened a clean `https://excalidraw.com` tab.

**Comment by kassiebowman** [comment]
Good catch on the new tab; I verified this morning if I go back to the first tab that was in the collaboration session and refresh, the new shapes are present in the library and I can use them like expected. It would be nice if there was some kind of indication to the user that this is what happened so they would know to refresh the original tab.

I haven't taken a look at the code yet, but is there a reason that adding a shape library has to open a new tab instead of adding the shapes to the already open tab?

**Comment by dwelle** [comment]
> I haven't taken a look at the code yet, but is there a reason that adding a shape library has to open a new tab instead of adding the shapes to the already open tab?

Not something that browsers would let us do. What we're doing currently is open excalidraw with a query link to the the library url which we then fetch on initialization and add to the library.

But we could certainly ensure the library is synced between all open tabs so that you don't have to refresh them.

I've also opened an issue to not persist collab scenes to localStorage so that opening new tabs won't load the last collaboration scene state. But the proposed behavior is slightly controversial, so not sure if it'll be implemented.
